_posts/2017-04-22-Annoying-Mastodon-Users-with-a-Bot.md

```diff
@@ -0,0 +1,212 @@
+---
+title: Annoying Mastodon Users with a Bot
+layout: post
+permalink:
+author: bobby_brb3 
+date: 2017-04-22
+---
+
+Because of the recent influx of Japanese users on Mastodon, I wrote a bot to
+translate their toots to English. Unfortunately, this annoyed a large portion of
+the Mastodon user base. If you're interested in learning a bit about writing a
+Mastodon bot, read on.
+
+<!--excerpt-->
+
+#### But Why?
+[Mastodon][mastodon], the GNU Social compatible microblogging platform,
+recently gained quite a bit of popularity. In the aftermath, a popular Japanese
+blog [published an article][ascii] describing the service. Suddenly, the small
+service was being flooded by excited Japanese users, sending out their "toots"
+in their native tongue. This caused several users to be very annoyed by their
+timeline suddenly being filled with Hiragana and the like.
+
+It's a bit odd that so many people were complaining about the Japanese users on
+Mastodon. Why should they care? Just continue using it like you have been, and
+ignore the toots you can't read.
+
+I decided to take a tongue-in-cheek approach, and create a translation bot that
+would reply to any toots in Japanese with a translated English version of the
+status.
+
+#### Translation API
+To get this working, I would need some way of translating the toots. I decided
+to go with [Microsoft's Translator API][translator-api]. It's not very accurate
+for Japanese (or at least, it doesn't seem to be), so I decided to poke fun at
+this by using a Clippy image as the bot's avatar.
+
+![Clippy](/images/blog/annoying-mastadon-users-with-a-bot/clippy.jpg)
+
+To follow along, setup an account with Microsoft, and get a token for the
+translator API.
+
+### Mastodon API
+
+Mastodon privdes a nice Ruby gem on their [github][mastodon-api]. It's
+[well documented][gem-docs], and works pretty nicely for this silly task. To
+connect to a Mastodon instance, you need to authenticate as a user via Oauth.
+This would normally be handled whenever a user connects to your "Mastodon app",
+but since we are just building a bot (and not acting on behalf of a user) we
+just need the Ruby script to use a token to toot as our translation bot's
+mastodon account.
+
+To setup the Oauth tokens:
+1. Create a mastodon account. I used [mastodon.cloud][mastodon-cloud] for my
+bot's instance.
+2. Create your secret token with this [this tool][tinysubversions]. 
+3. Hang onto the generated `access_token`.
+
+### Writing the Bot
+
+First, let's get a Gemfile in place:
+```ruby
+source 'https://rubygems.org'                                                                                                                                                                                                           
+
+gem 'mastodon-api', require: 'mastodon'                                          
+gem 'curb'        
+```
+
+Now it's bundle time: `$ bundle install`
+
+And we can setup the basics of our script.
```

Comment:
> Setup is a noun; set up is a verb.
English is worse than PHP sometimes, I swear.

Yea, that's a good observation. I'll roll a bit more _introduction_ into that bit.